modules/integration/src/test/scala/scala/cli/integration/PackageTestDefinitions.scala

```diff
@@ -469,10 +469,60 @@ abstract class PackageTestDefinitions(val scalaVersionOpt: Option[String])
     }
   }
 
-  if (!Properties.isWin && actualScalaVersion.startsWith("2.13"))
+  def libraryNativeTest(shared: Boolean = false): Unit = {
+    val fileName     = "simple.sc"
+    val nativeTarget = if (shared) "dynamic" else "static"
+    val inputs = TestInputs(
+      os.rel / fileName ->
+        s"""
+           |//> using platform scala-native
+           |//> using nativeTarget $nativeTarget
```

Comment:
@Gedochao miraculously, the tests are now green.

One issue I discovered though is that I can't seem to override this directive by passing `--native-target` in the command line.

This (from my side) would be the last thing I'd want to address with this PR

Reply:
Nice!
I can try looking at it sometime next week, it's not exactly easy to pass it all the way from the command line.
I'll let you know when I do.

Reply:
We can also go through the package type option (I think I had that originally), but it's a bit confusing with --native-target.

Reply:
nice!